packages/waas/src/auth.ts

```diff
@@ -171,10 +185,91 @@ export class SequenceWaaS {
   ) {
     this.config = defaultArgsOrFail(config)
     this.waas = new SequenceWaaSBase({ network: 1, ...config }, this.store, this.cryptoBackend, this.secureStoreBackend)
-    this.client = new WaasAuthenticator(this.config.rpcServer, this.fetch.bind(this))
+    this.client = new WaasAuthenticator(this.config.rpcServer, this._fetch)
+
     this.deviceName = new StoreObj(this.store, '@0xsequence.waas.auth.deviceName', undefined)
   }
 
+  _fetch = (input: RequestInfo, init?: RequestInit): Promise<Response> => {
+    if (this.signatureVerificationFailed) {
+      throw new Error('Signature verification failed')
+    }
+
+    if (this.cryptoBackend && this.config.disableHttpSignatureCheck !== true && init?.headers) {
+      const headers: { [key: string]: any } = {}
+
+      headers['Accept-Signature'] = 'sig=();alg="rsa-v1_5-sha256"'
+
+      init!.headers = { ...init!.headers, ...headers }
+    }
+
+    const response = fetch(input, init)
+
+    if (this.cryptoBackend && this.config.disableHttpSignatureCheck !== true) {
+      response.then(async r => {
```

Comment:
I think you need `return` here, no?

Reply:
so I did it this way to make sure I don't add time overhead and only check stuff to validate and leave response as it, I also do `const clone = r.clone()` because otherwise stream will be closed after first read and it can't be used down the line.

Reply:
But this means that the request will appear valid despite having an invalid signature, no? If the signature is invalid, then we treat all future requests as invalid, but why not the first invalid request in this case?

Reply:
shared more details on DM

Reply:
![Screenshot 2024-09-03 at 21 22 21](https://github.com/user-attachments/assets/b774baa2-cb37-495b-90e2-c709dfa6d51e)